app/storage/encrypted_storage.py

```diff
@@ -5,80 +5,48 @@
 from app import settings
 from app.cryptography.jwe_decryption import JWEDirDecrypter
 from app.cryptography.jwe_encryption import JWEDirEncrypter
-from app.storage.abstract_server_storage import AbstractServerStorage
+from app.storage.database_storage import DatabaseStorage
 from app.utilities.strings import to_bytes
 from app.utilities.strings import to_str
 
 logger = logging.getLogger(__name__)
 
 
-class EncryptedServerStorageDecorator(AbstractServerStorage):
+def generate_key(user_id, user_ik, pepper=settings.EQ_SERVER_SIDE_STORAGE_ENCRYPTION_KEY_PEPPER):
+    sha256 = hashlib.sha256()
+    sha256.update(to_str(user_id).encode('utf-8'))
+    sha256.update(to_str(user_ik).encode('utf-8'))
+    sha256.update(to_str(pepper).encode('utf-8'))
 
-    JSON_DATA_KEY = 'data'
+    # we only need the first 32 characters for the CEK
+    cek = sha256.hexdigest()[:32]
+    return to_bytes(cek)
 
-    def __init__(self, server_storage):
+
+class EncryptedStorage(DatabaseStorage):
```

Comment:
By removing the store and get API contract I don't get why we have the get_storage factory method. If the method signatures differ then surely the encrypted storage toggle off function won't work? It was in this state before we started work on it so has it ever worked without encryption?

Going further than this if there isn't a IS-A relationship do we disadvantage ourselves by tightly coupling the two classes and composition would be favoured here?